veps/sig-network/decouple-nad.md

```diff
@@ -0,0 +1,183 @@
+
+
+##### Pros
+- No backward compatibility breach
+
+##### Cons
+- Virt-operator currently only deploys internal components and does not reconcile 3rd party tools. Letting it deal with external components may architecturally be a slippery slope that may furthermore contribute to blurring responsibilities between KubeVirt, [CNAO](https://github.com/kubevirt/cluster-network-addons-operator) and [HCO](https://github.com/kubevirt/hyperconverged-cluster-operator). 
+It will be hard to explain why it's important to have virt-operator deploy network-resources-injector, but other network dependencies such as multus, sriov-operator/device-plugin, various CNIs etc.
+- Bloats the code base, just to replace a few lines of code.
+- "Swallowing" an external project somehow defeats the objective of decoupling dependencies. 
+
+> [!NOTE] The rest of this document assumes Option #1
+
+## API Examples
+
+### NetworkAttachmentDefinitions
+
+**SR-IOV Network:**
+```yaml
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: sriov-net-attach-def
+  annotations:
+    k8s.v1.cni.cncf.io/resourceName: intel.com/sriov_net
+spec:
+  config: |
+    {
+      "cniVersion": "0.3.1",
+      "type": "sriov",
+      "vlan": 100
+    }
+```
+
+
+### Expected virt-launcher Pod (After network-resources-injector)
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: virt-launcher-testvm-xxxxx
+  annotations:
+    k8s.v1.cni.cncf.io/networks: '[{"name":"sriov-net-attach-def","namespace":"default"}]'
+spec:
+  containers:
+  - name: compute
+    resources:
+      requests:
+        intel.com/sriov_net: "1"
+      limits:
+        intel.com/sriov_net: "1"
+```
+
+## Scalability
+
+The network-resources-injector approach provides several scalability advantages over the current implementation:
+- Caching of NADs avoids redundant API calls.
+- By default runs in 2 replicas, but can be scaled to run in multiple replicas, thus highly available.
+
+## Update/Rollback Compatibility
+
+Existing VMs in clusters opting in with the feature gate will have to deploy network-resources-injector before upgrade so that new VMIs and migrated VMs will be able to consume custom network resources. 
+In order to rollback, users would need to disable the FG. If VMs are already running and are experiencing connectivity issues since they have not been assigned with their required network resources, they would have to be restarted for rescheduling.
+
+## Functional Testing Approach
+
+Existing e2e tests already rely on mapping of custom resources and thus will validate regression.
+
+## Implementation History
+
+<!--
+For example:
+01-02-1921: Implemented mechanism for doing great stuff. PR: <LINK>.
+03-04-1922: Added support for doing even greater stuff. PR: <LINK>.
+-->
+
+## Graduation Requirements
+
+### Alpha
+  Will be introduced in KubeVirt v1.8, protected by the `DisableNADResourceInjection` FG. If the FG is set, KubeVirt will **not** map custom-resources. 
```

Comment:
How will this change be communicated?

Reply:
Communicating this deprecation via the mailing list gives the community time to adjust.